streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -347,15 +348,16 @@ public KafkaStreams(final TopologyBuilder builder,
 
         for (int i = 0; i < threads.length; i++) {
             threads[i] = new StreamThread(builder,
-                                          config,
-                                          clientSupplier,
-                                          applicationId,
-                                          clientId,
-                                          processId,
-                                          metrics,
-                                          time,
-                                          streamsMetadataState,
-                                          cacheSizeBytes);
+                    config,
+                    clientSupplier,
+                    applicationId,
+                    clientId,
+                    processId,
+                    metrics,
+                    time,
+                    streamsMetadataState,
+                    cacheSizeBytes);
+            threads[i].setStreamsKafkaClient(streamsKafkaClient);
```

Comment:
Ok. Should I go ahead with the constructor approach or are we still deliberating on alternative approaches which are getting discussed ?